===== Patch instructions generated by SAVER =====
[+] { Insert: if ({ index_notify:n$18 == { 0 } }) free(*(index_notify:obj)) at 16 (line 445, column 8) }
===== Actual patch in diff format =====
@@ -443,6 +443,7 @@ index_notify (p11_index *index,
 		obj->handle = handle;
 		obj->attrs = removed;
 		if (!p11_dict_set (index->changes, &obj->handle, obj))
+			free(obj);
 			return_if_reached ();
 	}
 }
